File: exercism/javascript/binary/binary.js
const { map, add } = require('lodash/fp')


class Binary {
  constructor (bin) {
    this.digits = bin.match(/[^10]/) ? [] : map(parseInt, bin.split(''))
  }
  toDecimal () {
    return this.digits.reverse().map((dig, i) => dig << i).reduce(add, 0)
  }
}

module.exports = Binary

File: exercism/c/binary/src/binary.c
#include "binary.h"

int convert(char *digits) {
  int total = 0;
  char d;

  while ((d = *digits++) != '\0') {
    if (d != '0' && d != '1')
      return INVALID;
    total = (total << 1) + (d - '0');
  }
  return total;
}
